### PR TITLE
Add option to setup unprivileged compilation of Vespa.

### PR DESCRIPTION
**Pull request by geirst**
@toregge and @vekterli please review

### CODE REVIEW

**Review by toregge — APPROVED** [review]
LGTM

File: bootstrap-cmake.sh [hunk 1]
```diff
@@ -33,3 +61,4 @@ cmake3 \
     ${EXTRA_CMAKE_ARGS} \
     -DVESPA_LLVM_VERSION=$VESPA_LLVM_VERSION \
     "${SOURCE_DIR}"
+fi
```
Comment (vekterli) [inline bootstrap-cmake.sh]: For code de-duping consider creating a `$VESPA_INSTALL_PREFIX` variable dependent on `$UNPRIVILEGED` and substituting that for `/opt/vespa` vs `$HOME/vespa` and also possibly appending VESPA_USER/VESPA_UNPRIVILEGED stuff to `EXTRA_CMAKE_ARGS`

**Comment by geirst** [comment]
@toregge and @vekterli PTAL

**Review by toregge — APPROVED** [review]
LGTM

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**